ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java

```diff
@@ -54,50 +54,171 @@ public String getVersion() {
-        RaoComputationResult raoComputationResult = buildRaoComputationResult(crac);
-        return CompletableFuture.completedFuture(raoComputationResult);
+        preOptimSensitivityAnalysisResult = SystematicSensitivityAnalysisService.runAnalysis(network, crac, computationManager);
+        postOptimSensitivityAnalysisResult = preOptimSensitivityAnalysisResult;
+        SystematicSensitivityAnalysisResult midOptimSensitivityAnalysisResult;
+        List<RemedialActionResult> oldRemedialActionsResult = new ArrayList<>();
+
+        int iterationsLeft = MAX_ITERATIONS;
+
+        double oldScore = getMinMargin(crac, preOptimSensitivityAnalysisResult);
+        String originalNetworkVariant = network.getVariantManager().getWorkingVariantId();
+        LinearRaoModeller linearRaoModeller = createLinearRaoModeller(crac, network, preOptimSensitivityAnalysisResult);
+        linearRaoModeller.buildProblem();
+        List<AbstractPostProcessor> postProcessorList = new ArrayList<>();
+        postProcessorList.add(new RaoResultPostProcessor());
+
+        RaoComputationResult raoComputationResult;
+        List<RemedialActionResult> newRemedialActionsResult;
+
+        while (iterationsLeft > 0) {
+            raoComputationResult = linearRaoModeller.solve();
+            if (raoComputationResult.getStatus() == RaoComputationResult.Status.FAILURE) {
+                return CompletableFuture.completedFuture(raoComputationResult);
+            }
+
+            newRemedialActionsResult = raoComputationResult.getPreContingencyResult().getRemedialActionResults();
+            //TODO: manage CRA
+            if (sameRemedialActionResultLists(newRemedialActionsResult, oldRemedialActionsResult)) {
+                break;
+            }
+
+            createAndSwitchToNewVariant(network, originalNetworkVariant);
+            applyRAs(crac, network, newRemedialActionsResult);
+            midOptimSensitivityAnalysisResult = SystematicSensitivityAnalysisService.runAnalysis(network, crac, computationManager);
+            double newScore = getMinMargin(crac, midOptimSensitivityAnalysisResult);
+            if (newScore < oldScore) {
+                // TODO : limit the ranges
+                LOGGER.warn("Linear Optimization found a worse result after an iteration: from {} to {}", oldScore, newScore);
+                break;
+            }
+
+            postOptimSensitivityAnalysisResult = midOptimSensitivityAnalysisResult;
+            oldScore = newScore;
+            oldRemedialActionsResult = newRemedialActionsResult;
+            linearRaoModeller.updateProblem(midOptimSensitivityAnalysisResult);
+            iterationsLeft -= 1;
+        }
+
+        return CompletableFuture.completedFuture(buildRaoComputationResult(crac, oldRemedialActionsResult, oldScore));
+    }
+
+    LinearRaoModeller createLinearRaoModeller(Crac crac,
```

Comment:
Add a comment to mention it's defined for testing purpose

Reply:
ok